.ci/test.sh

```diff
@@ -13,6 +13,11 @@ ARCH=$(uname -m)
 
 LGB_VER=$(head -n 1 "${BUILD_DIRECTORY}/VERSION.txt")
 
+# create the artifact upload directory if it doesn't exist yet
+if [[ -n "${BUILD_ARTIFACTSTAGINGDIRECTORY:-}" ]]; then
+    mkdir -p "${BUILD_ARTIFACTSTAGINGDIRECTORY}"
+fi
```

Comment:
Proposing this same change in #7024 too... unlike Azure DevOps, no dedicated artifacts directory is automatically created.